src/Wolk.cpp

```diff
@@ -43,22 +49,52 @@ WolkBuilder Wolk::newBuilder(Device device)
     return WolkBuilder(device);
 }
 
+template <typename T> void Wolk::addSensorReading(const std::string& reference, T value, unsigned long long rtc)
+{
+    addSensorReading(reference, std::to_string(value), rtc);
+}
+
 template <> void Wolk::addSensorReading(const std::string& reference, std::string value, unsigned long long rtc)
 {
-    if (rtc == 0)
+    auto sensorReading = std::make_shared<SensorReading>(value, reference, rtc != 0 ? rtc : Wolk::currentRtc());
+    addToCommandBuffer(
+      [=]() -> void { m_persistence->putSensorReading(sensorReading->getReference(), sensorReading); });
+}
+
+template <typename T>
+void Wolk::addSensorReading(const std::string& reference, std::initializer_list<T> values, unsigned long long int rtc)
```

Comment:
Are both of these needed (initializer_list and vector) and if so can this code duplication be avoided?